internal/clients/builder.go

```diff
@@ -109,9 +109,14 @@ func Build(ctx context.Context, builder ClientBuilder) (*Client, error) {
 		return nil, fmt.Errorf("building account: %+v", err)
 	}
 
-	managedHSMAuth, err := auth.NewAuthorizerFromCredentials(ctx, *builder.AuthConfig, builder.AuthConfig.Environment.ManagedHSM)
-	if err != nil {
-		return nil, fmt.Errorf("unable to build authorizer for Managed HSM API: %+v", err)
+	var managedHSMAuth auth.Authorizer
+	if _, ok := builder.AuthConfig.Environment.ManagedHSM.ResourceIdentifier(); ok {
```

Comment:
the ResourceIdentifier is cloud agnostic - the Environment is cloud-specific, so we'll want to use that:

```suggestion
	if _, ok := builder.AuthConfig.Environment.ManagedHSM.Environment(); ok {
```

Reply:
The Api interface has no `Environment()` method

https://github.com/hashicorp/terraform-provider-azurerm/blob/868867d3674bf424d4feb5b34aeb663916ed72c0/vendor/github.com/hashicorp/go-azure-sdk/sdk/environments/interfaces.go#L6-L12